### PR TITLE
Decouple components so they can be shared between profiles

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- load your profile and another user's profile
- the profiles should appear as before

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7834/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7835/

File: jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js [hunk 1]
```diff
@@ -83,10 +86,8 @@ class UserProfileModelList extends React.Component {
           return model.owner === extUser;
         });
       }
-      if (modelList.length) {
-        broadcastStatus('ok');
-      } else {
-        broadcastStatus('empty');
+      if (modelList && modelList.length > 0) {
+        this.props.setHasEntities(true);
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js]: Do we ever set this to false? I mean, even if we do, I am not sure about one single component setting has entities to falso for all the others. So maybe simplify this call like `this.props.notifyHasEntities()` ?

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 2]
```diff
@@ -28,6 +35,15 @@ class UserProfile extends React.Component {
   }
 
   /**
+    Set whether the profile has entities.
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: Please consider the change above, and also describe better how and where this is set.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 3]
```diff
@@ -53,20 +69,31 @@ class UserProfile extends React.Component {
   }
 
   /**
+    Generate the empty state.
+
+    @method _generateContent
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: This is wrong and you can just remove this line.

**Comment by frankban** [comment]
QA is suboptimal. On a not amazingly fast browser, every time you click to go to a profile page the empty profile is displayed, and it disappears when the calls to get charms, bundle etc return and there are contents. But initially it's just confusing. So I'd suggest that the components collaborate to sigal that the profile is ready, and only at that point we can decide whether to display the empty profile.

**Comment by hatched** [comment]
After discussing with @frankban we've determined that we will still need a scaleable watcher even in the new profile view for the summary page. And for situations where we are rendering these components without the watcher, they can simply be updated to call a noop broadcast method if none is provided.

**Comment by huwshimi** [comment]
As discussed with @hatched we'll need to do this work eventually so I've fixed the loading issue. The sections now show the spinner and if all sections are empty then the empty profile state is displayed.

**Comment by huwshimi** [comment]
Continued in #3054.

**Pull request closed without merging**